src/libpsc/psc_output_fields/fields_item_fields.hxx

```diff
@@ -77,76 +77,129 @@ private:
 };
 
 // ======================================================================
-// Item_dive
+// psc::item::
 
 namespace psc
 {
 namespace item
 {
 
-template <typename E>
-inline auto div_yz(const E& flds, const Grid_t& grid)
+namespace
 {
-  Int3 bnd = {(flds.shape(0) - grid.domain.ldims[0]) / 2,
-              (flds.shape(1) - grid.domain.ldims[1]) / 2,
-              (flds.shape(2) - grid.domain.ldims[2]) / 2};
-
-  auto dxyz = grid.domain.dx;
+// FIXME is this namespace appropriate?
+auto s0 = _s(1, _);
+auto sm = _s(_, -1);
 
-  auto s0 = _s(1, _);
-  auto sm = _s(_, -1);
-
-  auto res = gt::empty<gt::expr_value_type<E>, gt::expr_space_type<E>>(
-    {grid.ldims[0], grid.ldims[1], grid.ldims[2], 1, grid.n_patches()});
-
-  auto _flds =
-    flds.view(_all, _s(-1 + bnd[1], -bnd[1]), _s(-1 + bnd[2], -bnd[2]));
+template <typename GT, typename... SLICES>
+auto viewFromSlices(GT& gt, gt::gslice slices[], SLICES... moreSlices)
+{
+  return gt.view(slices[0], slices[1], slices[2], moreSlices...);
+}
 
-  res.view(_all, _all, _all, 0) =
-    (_flds.view(_all, s0, s0, 1) - _flds.view(_all, sm, s0, 1)) / dxyz[1] +
-    (_flds.view(_all, s0, s0, 2) - _flds.view(_all, s0, sm, 2)) / dxyz[2];
+template <typename GT1, typename GT2>
+void plusEquals(GT1&& lhs, GT2&& rhs)
+{
+  lhs = lhs + rhs;
+}
```

Comment:
I think this is something that really needs to be fixed on the gtensor side (making += work). Not really a fan of this helper. I do think this needs some `std::forward`, but in fact that may not be possible for `lhs` since it's used twice.